src/app/store/nav.reducer.ts

```diff
@@ -19,14 +38,37 @@ export function navReducer(state = NavState, action: NavActions): NavInterface {
       const routeDetails = (action as SetRouteAction).payload.routeDetails;
       return { ...state, routeDetails: routeDetails };
     }
-    case navActionTypes.SET_ACTIVE_NAV: {
-      const isActiveNav = (action as SetActiveNavAction).payload.isActiveNav;
-      return { ...state, isActiveNav };
+    case navActionTypes.SET_PHRAZE_STATE: {
+      const phrazeState = (action as SetPhrazeStateAction).payload.phrazeState;
+      return { ...state, phrazeState };
     }
     case navActionTypes.SET_SHOW_SEARCH: {
       const isShowSearch = (action as SetShowSearchAction).payload.isShowSearch;
       return { ...state, isShowSearch };
     }
+    case navActionTypes.SET_CURRENT_POSITION: {
+      const currentPosition = (action as SetCurrentPositionAction).payload.currentPosition;
+      const previousPosition = state.currentPosition;
+      const previousPositionTimeStamp = state.currentPositionTimeStamp;
+      const currentPositionTimeStamp = Date.now();
+      return { ...state, currentPosition, previousPositionTimeStamp, currentPositionTimeStamp, previousPosition };
+    }
+    case navActionTypes.SET_NEXT_WAYPOINT_INDEX: {
+      const nextWaypointIndex = (action as SetNextWaypointIndexAction).payload.nextWaypointIndex;
+      return { ...state, nextWaypointIndex: nextWaypointIndex, isNextWpNotified: false };
+    }
+    case navActionTypes.SET_NEXT_WAYPOINT_DISTANCE: {
+      const nextWaypointDistance = (action as SetNextWaypointDistanceAction).payload.nextWaypointDistance;
+      return { ...state, nextWaypointDistance: nextWaypointDistance };
+    }
+    case navActionTypes.SET_DISTANCE_TO_ENDPOINT: {
```

Comment:
This can be part of SET_NEXT_WAYPOINT_DISTANCE as overall distance doesn't change without distance to next way point